x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -111,6 +117,8 @@
     private volatile Map<String, Set<String>> changedBuckets;
     private volatile Map<String, Object> changedBucketsAfterKey;
 
+    private long lastCheckpointCleanup = 0L;
```

Comment:
```suggestion
    private volatile long lastCheckpointCleanup = 0L;
```
Since it is written and read by different threads. Shouldn't be an issue (i think), but Java concurrency can get funky.